packages/web/src/SessionBasedSampler.ts

```diff
@@ -72,20 +71,22 @@ export class SessionBasedSampler implements Sampler {
 		traceId: string,
 		spanName: string,
 		spanKind: SpanKind,
-		attributes: SpanAttributes,
+		attributes: Attributes,
 		links: Link[],
 	): SamplingResult {
 		// Implementation based on @opentelemetry/core TraceIdRatioBasedSampler
 		// but replacing deciding based on traceId with sessionId
 		// (not extended from due to private methods)
-		const currentSession = getRumSessionId({ useLocalStorage: this.useLocalStorageForSessionMetadata })
-		if (this._currentSession !== currentSession) {
-			this._currentSessionSampled = this._accumulate(currentSession) < this._upperBound
-			this._currentSession = currentSession
+
+		// TODO: we are guaranteed to have a session here, so maybe error explicitly if that's not the case?
+
+		const currentSessionId = getOrInitShadowSession().id;
+		if (this._currentSessionId !== currentSessionId) {
```

Comment:
the shadow (or active) session's ID is stable for the same period of time as a normal session would, and session samplers check similarly returns the same result for the same session ID